drivers/sensors/cirque_pinnacle.c

```diff
@@ -265,10 +300,18 @@ pinnacle_data_t cirque_pinnacle_read_data(void) {
 #else
     // Decode data for relative mode
     // Registers 0x16 and 0x17 are unused in this mode
-    result.buttons    = data[0] & 0x07; // bit0 = primary button, bit1 = secondary button, bit2 = auxilary button, if Taps enabled then also software-recognized taps are reported
-    result.xDelta     = data[1];
-    result.yDelta     = data[2];
-    result.wheelCount = data[3];
+    result.buttons = data[0] & 0x07; // Only three buttons are supported
+    if ((data[0] & 0x10) && data[1] != 0) {
+        result.xDelta = -((int16_t)256 - (int16_t)(data[1]));
+    } else {
+        result.xDelta = data[1];
+    }
+    if ((data[0] & 0x20) && data[2] != 0) {
+        result.yDelta = ((int16_t)256 - (int16_t)(data[2]));
+    } else {
+        result.yDelta = -((int16_t)data[2]);
+    }
```

Comment:
```suggestion
result.xDelta = (int8_t)data[1];
result.yDelta = -((int8_t)data[2]);
```
These are simply two's complement.
Also change `xDelta` and `yDelta` to `int8_t` in `pinnacle_data_t`.

Reply:
The change will not work, the sign is in data[0], so the range is -255 to 255.

Reply:
Got it, thanks. I didn't see mention of  X & Y delta range in Cirque's datasheet and not all PS/2 protocol pages mention it's 9-bits.

Reply:
It's in Table 10 (Relative Position Registers Byte Format) in https://www.dropbox.com/s/to7pxwesph71666/IC-DS-150408%20Pinnacle%20Specification.pdf?dl=0

Reply:
It just has an example of two's complement under that table, no explicit mention of the words "9-bit", or "-255 ~ +255". So my dumb brain couldn't understand that the sign bits were part of the two's complement...